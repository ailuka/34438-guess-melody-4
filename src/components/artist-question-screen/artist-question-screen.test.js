import React from "react";
import renderer from "react-test-renderer";
import ArtistQuestionScreen from "./artist-question-screen.jsx";

const question = {
  type: `artist`,
  song: {
    artist: `Lorde`,
    src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  },
  answers: [{
    picture: `https://api.adorable.io/avatars/128/1`,
    artist: `Пелагея`,
  }, {
    picture: `https://api.adorable.io/avatars/128/2`,
    artist: `Краснознаменная дивизия имени моей бабушки`,
  }, {
    picture: `https://api.adorable.io/avatars/128/3`,
    artist: `Lorde`,
  }],
};

describe(`src/artist-question-screen.jsx`, () => {
  describe(`when there is no questions`, () => {
    it(`should render`, () => {
      const tree = renderer.create(
          <ArtistQuestionScreen
            question={question}
            onAnswerClick={() => null}
          />
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
