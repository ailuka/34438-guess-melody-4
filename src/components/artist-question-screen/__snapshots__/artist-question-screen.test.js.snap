// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/artist-question-screen.jsx when there is a question should render 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="Пелагея"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/1"
        />
        Пелагея
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Краснознаменная дивизия имени моей бабушки"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/2"
        />
        Краснознаменная дивизия имени моей бабушки
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        onChange={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Lorde"
          className="artist__picture"
          src="https://api.adorable.io/avatars/128/3"
        />
        Lorde
      </label>
    </div>
  </form>
</section>
`;
